saleor/csv/utils/export.py

```diff
@@ -84,13 +92,68 @@ def get_product_queryset(scope: Dict[str, Union[str, dict]]) -> "QuerySet":
     elif "filter" in scope:
         queryset = ProductFilter(data=scope["filter"], queryset=queryset).qs
 
-    queryset = queryset.order_by("pk").prefetch_related(
-        "attributes", "variants", "collections", "images", "product_type", "category"
-    )
+    queryset = queryset.order_by("pk")
 
     return queryset
 
 
+def export_products_in_batches(
+    queryset: "QuerySet",
+    export_info: Dict[str, list],
+    export_fields: Set[str],
+    headers: List[str],
+    csv_headers_mapping: Dict[str, str],
+    delimiter: str,
+    export_file: ExportFile,
+    file_name: str,
+    file_type: str,
+):
+    products_count = queryset.count()
+    warehouses = export_info.get("warehouses")
+    attributes = export_info.get("attributes")
+
+    create_file = True
+    counter = 0
+    while counter < products_count:
+        qs = queryset[counter : counter + BATCH_SIZE]
```

Comment:
Just for the record - `count` and slicing the queryset (database `OFFSET`) will be expensive and slow for large querysets. We agreed with @IKarbowiak to try to refactor this to use cursor-based approach as we do for pagination.